### DISCUSSION TITLE
Backdash - new C# rollback netcode library

### DISCUSSION THREAD

**Discussion by lucasteles** [MonoGame/MonoGame#8245] [General]
I've been working on a dotnet library for [Rollback Netcode](https://en.wikipedia.org/wiki/Netcode#Rollback), widely used by fighting games to accomplish a better P2P multiplayer experience

I based my implementation on [GGPO](https://github.com/pond3r/ggpo) which was the pioneer in this type of netcode

More about why it is good and how it works [here](https://www.youtube.com/watch?v=0NLe4IpdS1w) and [here](https://words.infil.net/w02-netcode.html).

It is very early and needs more testing and docs. but I believe it looks good enough to share 😅

### https://github.com/lucasteles/Backdash

Monogame local network demo:
https://www.youtube.com/watch?v=JYf2MemyJaY

Monogame online lobby demo:
https://www.youtube.com/watch?v=LGM_9XfzRUI

**Comment by ThomasFOG** [comment]
Very interesting and nice work!
